gadgets/fsnotify/test/unit/fsnotify_test.go

```diff
@@ -0,0 +1,185 @@
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/gadgetrunner"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/testing/utils"
+)
+
+type ExpectedFsnotifyEvent struct {
+	Timestamp string `json:"timestamp"`
+
+	Type string `json:"type"`
+
+	TraceeProc ebpftypes.Process `json:"tracee_proc"`
+	TracerProc ebpftypes.Process `json:"tracer_proc"`
+
+	TraceeMntnsId uint64 `json:"tracee_mntns_id"`
+	TracerMntnsId uint64 `json:"tracer_mntns_id"`
+
+	TraceeUId uint32 `json:"tracee_uid"`
+	TraceeGId uint32 `json:"tracee_gid"`
+	TracerUId uint32 `json:"tracer_uid"`
+	TracerGId uint32 `json:"tracer_gid"`
+
+	Prio uint32 `json:"prio"`
+
+	FaMask uint32 `json:"fa_mask"`
+	IMask  uint32 `json:"i_mask"`
+
+	FaType     string `json:"fa_type"`
+	FaPId      uint32 `json:"fa_pid"`
+	FaFlags    uint32 `json:"fa_flags"`
+	FaFFlags   uint32 `json:"fa_f_flags"`
+	FaResponse string `json:"fa_response"`
+
+	IWd     int32  `json:"i_wd"`
+	ICookie uint32 `json:"i_cookie"`
+	IIno    uint32 `json:"i_ino"`
+	IInoDir uint32 `json:"i_ino_dir"`
+
+	Name string `json:"name"`
+}
+
+type testDef struct {
+	runnerConfig  *utilstest.RunnerConfig
+	generateEvent func() (string, error)
+	validateEvent func(t *testing.T, info *utilstest.RunnerInfo, filename string, events []ExpectedFsnotifyEvent)
+}
+
+func TestFsnotifyGadget(t *testing.T) {
+	gadgettesting.InitUnitTest(t)
+	runnerConfig := &utilstest.RunnerConfig{}
+
+	testCases := map[string]testDef{
+		"captures_inotify_event": {
+			runnerConfig:  runnerConfig,
+			generateEvent: generateEvent,
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, filename string, events []ExpectedFsnotifyEvent) {
+				utilstest.ExpectAtLeastOneEvent(func(info *utilstest.RunnerInfo, pid int) *ExpectedFsnotifyEvent {
+					return &ExpectedFsnotifyEvent{
+						Type: "inotify",
+
+						IMask: 134217732, // 134217732 = 0x08000004 = FS_ATTRIB | FS_EVENT_ON_CHILD
+						Name:  filename,
+
+						Timestamp: utils.NormalizedStr,
+
+						TraceeMntnsId: utils.NormalizedInt,
+						TracerMntnsId: utils.NormalizedInt,
+
+						FaType:     utils.NormalizedStr,
+						FaResponse: utils.NormalizedStr,
+
+						IWd:     utils.NormalizedInt,
+						IIno:    utils.NormalizedInt,
+						IInoDir: utils.NormalizedInt,
+					}
+				})(t, info, 0, events)
+			},
+		},
+	}
+	for name, testCase := range testCases {
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			var filename string
+			runner := utilstest.NewRunnerWithTest(t, testCase.runnerConfig)
+
+			normalizeEvent := func(event *ExpectedFsnotifyEvent) {
+				utils.NormalizeString(&event.Timestamp)
+
+				utils.NormalizeProc(&event.TraceeProc)
+				utils.NormalizeProc(&event.TracerProc)
+				utils.NormalizeInt(&event.TraceeMntnsId)
+				utils.NormalizeInt(&event.TracerMntnsId)
+
+				utils.NormalizeInt(&event.TraceeUId)
+				utils.NormalizeInt(&event.TraceeGId)
+				utils.NormalizeInt(&event.TracerUId)
+				utils.NormalizeInt(&event.TracerGId)
```

Comment:
Can't we test then too? The runner allows to use a different uid / gid. See fields on utilstest.RunnerConfig.

Reply:
Yeah, I tried using different uid/gid values, but neither the tracer nor the tracee has these values. Only runnerInfo has the new, different uid/gid.

Need to debug why we’re unable to retrieve those values in tracer/tracee.